server/py/services/api/crud/runtimes/nuclio/function.py

```diff
@@ -510,28 +510,27 @@ def _resolve_and_set_nuclio_runtime(
 ):
     nuclio_runtime = (
         function.spec.nuclio_runtime
-        or services.api.crud.runtimes.nuclio.helpers.resolve_nuclio_runtime_python_image(
-            mlrun_client_version=client_version, python_version=client_python_version
-        )
+        # TODO: Uncomment when the function is aligned with the supported python versions
+        # or services.api.crud.runtimes.nuclio.helpers.resolve_nuclio_runtime_python_image(
+        #     mlrun_client_version=client_version, python_version=client_python_version
+        # )
+        or mlrun.mlconf.default_nuclio_runtime
     )
 
-    # For backwards compatibility, we need to adjust the runtime for old Nuclio versions
-    # TODO: remove in 1.8, default to 3.9
-    if services.api.crud.runtimes.nuclio.helpers.is_nuclio_version_in_range(
-        "0.0.0", "1.6.0"
+    if mlrun.runtimes.nuclio.function.validate_nuclio_version_compatibility(
+        "1.14.1",
     ) and nuclio_runtime in [
+        "python:3.6",
         "python:3.7",
         "python:3.8",
```

Comment:
worth doing the migration automatically -> `python:3.6` -> `default` as client using mlrun<>nuclio wont set the version explicitly but rather use nuclio which defaults its runtime to whatever legacy used to have.

Reply:
you mean instead of failing, replace the version to the mlrun default version?
1. We only fail if the user set the python version excplicitly - automatically changing it might cause unwanted issues.
2. Having only a warning will not be visible, because this function is in the server.

Reply:
discussed offline - do the validation/enrichment only after setting the base image as if we set the base image to `mlrun/mlrun` there is no real meaning for the python version here.